reader-osm/src/main/java/com/graphhopper/reader/osm/OSMHandler.java

```diff
@@ -0,0 +1,38 @@
+package com.graphhopper.reader.osm;
+
+import com.graphhopper.reader.ReaderElement;
+import com.graphhopper.reader.ReaderNode;
+import com.graphhopper.reader.ReaderRelation;
+import com.graphhopper.reader.ReaderWay;
+
+/**
+ * Interface with some callback methods to called with each OSM object read from an OSM input file.
+ *
+ * @author Michael Reichert
+ */
+public interface OSMHandler {
+    /**
+     * Called for each node, way or relation read from the input file
+     */
+    public void osm_object(ReaderElement element);
```

Comment:
Can you rename this method to e.g. osmObject to match Java formatting standard? Also I cannot find the place where this method is called - is this accidentally missing?

Reply:
Sorry, I am doing to much C++ with the Libosmium code style. :-) I will change this.

`osm_object` is used to count the number of parsed objects in `OSMReader::HandlerPass2`. The callback is called for each node, way and relation. (By the way, Libosmium estimates the progress by looking at the number of read bytes vs. the length of the file)